docs/tutorial/expressjs.rst

```diff
@@ -0,0 +1,430 @@
+``expressjs`` service running inside the container.
+We should expect to see something similar to this:
+
+.. terminal::
+
+    app@0.0.0 start
+    node ./bin/www
+    GET / 200 62.934 ms - 170
+
+We can also choose to follow the logs by using the ``-f`` option with the
+``pebble logs`` command above. To stop following the logs, press :kbd:`Ctrl` +
+:kbd:`C`.
+
+Cleanup
+~~~~~~~
+
+Now we have a fully functional rock for a ExpressJS application! This concludes
+the first part of this tutorial, so we'll stop the container and remove the
+respective image for now:
+
+.. literalinclude:: code/expressjs/task.yaml
+    :language: bash
+    :start-after: [docs:stop-docker]
+    :end-before: [docs:stop-docker-end]
+    :dedent: 2
+
+Chisel the rock
+===============
+
+This is an optional but recommended step, especially if we're looking to
+deploy the rock into a production environment. With :ref:`chisel_explanation`
+we can produce lean and production-ready rocks by getting rid of all the
+contents that are not needed for the ExpressJS application to run. This results
+in a much smaller rock with a reduced attack surface.
+
+.. note::
+    It is recommended to run chiselled images in production. For development,
+    we may prefer non-chiselled images as they will include additional
+    development tooling (such as for debugging).
+
+The first step towards chiselling the rock is to ensure we are using a
+``bare`` :ref:`base <bases_explanation>`.
+In ``rockcraft.yaml``, change the ``base`` to ``bare`` and add
+``build-base: ubuntu@24.04``:
+
+.. literalinclude:: code/expressjs/task.yaml
+    :language: bash
+    :start-after: [docs:change-base]
+    :end-before: [docs:change-base-end]
+    :dedent: 2
+
+.. note::
+    The ``sed`` command replaces the current ``base`` in ``rockcraft.yaml`` with
+    the ``bare`` base. Note that ``build-base`` is also required when using the
+    ``bare`` base.
+
+So that we can compare the size after chiselling, open the ``rockcraft.yaml``
+file and change the ``version`` (e.g. to ``0.1-chiselled``). Pack the rock with
+the new ``bare`` :ref:`base <bases_explanation>`:
+
+.. literalinclude:: code/expressjs/task.yaml
+    :language: bash
+    :start-after: [docs:chisel-pack]
+    :end-before: [docs:chisel-pack-end]
+    :dedent: 2
+
+As before, verify that the new rock was created:
+
+.. literalinclude:: code/expressjs/task.yaml
+    :language: bash
+    :start-after: [docs:ls-bare-rock]
+    :end-before: [docs:ls-bare-rock-end]
+    :dedent: 2
+
+We'll verify that the new ExpressJS rock is now approximately **15% smaller**
+in size! And that's just because of the simple change of ``base``.
+
+And the functionality is still the same. As before, we can confirm this by
+running the rock with Docker
+
+.. literalinclude:: code/expressjs/task.yaml
+    :language: text
+    :start-after: [docs:docker-run-chisel]
+    :end-before: [docs:docker-run-chisel-end]
+    :dedent: 2
+
+and then using the same ``curl`` request:
```

Comment:
```suggestion
running the rock with Docker:

.. literalinclude:: code/expressjs/task.yaml
    :language: text
    :start-after: [docs:docker-run-chisel]
    :end-before: [docs:docker-run-chisel-end]
    :dedent: 2

And using the same curl request:
```

Reply:
It would likely be simpler to combine these steps.

Reply:
Chiselling step has been removed in favor of the `bare` base default :)